core/logger/sentry.go

```diff
@@ -0,0 +1,325 @@
+package logger
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/core/static"
+)
+
+const SentryFlushDeadline = 5 * time.Second
+
+func init() {
+	// If SENTRY_ENVIRONMENT is set, it will override everything. Otherwise infers from CHAINLINK_DEV.
+	var sentryenv string
+	if env := os.Getenv("SENTRY_ENVIRONMENT"); env != "" {
+		sentryenv = env
+	} else if os.Getenv("CHAINLINK_DEV") == "true" {
+		sentryenv = "dev"
+	} else {
+		sentryenv = "prod"
+	}
+	// If SENTRY_DSN is set, it will override everything. Otherwise static.SentryDSN will be used.
+	// If neither are set, sentry is disabled.
+	var sentrydsn string
+	if dsn := os.Getenv("SENTRY_DSN"); dsn != "" {
+		sentrydsn = dsn
+	} else {
+		sentrydsn = static.SentryDSN
+	}
+	// If SENTRY_RELEASE is set, it will override everything. Otherwise, static.Version will be used.
+	var sentryrelease string
+	if release := os.Getenv("SENTRY_RELEASE"); release != "" {
+		sentryrelease = release
+	} else {
+		sentryrelease = static.Version
+	}
+	err := sentry.Init(sentry.ClientOptions{
+		// AttachStacktrace is needed to send stacktrace alongside panics
+		AttachStacktrace: true,
+		Dsn:              sentrydsn,
+		Environment:      sentryenv,
+		Release:          sentryrelease,
+		// Enable printing of SDK debug messages.
+		// Uncomment line below to debug sentry
+		// Debug: true,
+	})
+	if err != nil {
+		log.Fatalf("sentry.Init: %s", err)
+	}
+}
+
+//TODO could include sentry event IDs with log lines: .With("sentryEvent",*EventID)
```

Comment:
@samsondav did you consider including the returned `Capture*() EventID`s ? Would it be worthwhile for the node ops to have this ID for reference?

Reply:
Follow up story: https://app.shortcut.com/chainlinklabs/story/22266/sentry-log-eventids